lib/assets/javascripts/cartodb/models/filter.js

```diff
@@ -587,7 +591,8 @@ cdb.admin.models.Filters = Backbone.Collection.extend({
       return f.getSQLCondition();
     })
 
-    var sql = _(sqls).compact().join(' AND ');
+    var operator = this.dataLayer.get('sql_or_toggle') ? ' OR ' : ' AND ';
```

Comment:
why not "filter_operator" so it contains "AND", "OR" or whatever you want to use